tests/composite_pk/test_filter.py

```diff
@@ -469,6 +469,26 @@ def test_outer_ref_pk(self):
                 queryset = Comment.objects.filter(**{f"id{lookup}": subquery})
                 self.assertEqual(queryset.count(), expected_count)
 
+    def test_outer_ref_pk_select_pk(self):
+        subquery = Subquery(User.objects.filter(pk=OuterRef("pk")).values("pk"))
+        tests = [
+            ("", 2),
+            ("__gt", 0),
+            ("__gte", 2),
+            ("__lt", 0),
+            ("__lte", 2),
+        ]
+        for lookup, expected_count in tests:
+            with self.subTest(f"id{lookup}"):
+                queryset = Comment.objects.filter(**{f"pk{lookup}": subquery})
```

Comment:
This fails on Oracle as mentioned on the ticket, but I'm wondering if that's a symptom of a deeper problem that we should fix first & separately?